spec/lib/open_food_network/products_renderer_spec.rb

```diff
@@ -49,6 +49,7 @@ module OpenFoodNetwork
       end
 
       it "doesn't return products not in stock" do
+        variant.update_attribute(:on_demand, false)
         variant.update_attribute(:count_on_hand, 0)
```

Comment:
you could use `update_attributes` here with both of them.

Reply:
I tried with `variant.update_attribute(on_demand: false, count_on_hand: 0)` but it still fails with the same error

Reply:
Note the extra s in `update_attributeS`. Did you try this? it's weird anyway, because it only changes from 2 UPDATE db statements to a single one.

Reply:
I tried with `attributes`*, sorry I typed it wrong in GH

Reply:
Oh, I figured. Anyway, it's not a big deal we just want to fix this failure.

Reply:
you need this commit to make update_attributes work on on_demand, right?
https://github.com/openfoodfoundation/openfoodnetwork/pull/3132/commits/6cdaaa9e7d7e19dbcd997c29ba2b9648a229e82c

Reply:
also, just a heads up, I think we have a few of these cases in the broken tests, the on_demand defaults to true in the factory but some tests expect it to be false to test stock levels.